Composer/packages/server/src/models/asset/assetManager.ts

```diff
@@ -92,58 +97,86 @@ export class AssetManager {
     return ref;
   }
 
-  private async getRemoteTemplate(template: BotTemplate, destinationPath: string) {
-    // install package
-    if (template.package) {
-      const { stderr: initErr } = await execAsync(`dotnet new -i ${template.package.packageName}`);
-      if (initErr) {
-        throw new Error(initErr);
+  public async copyRemoteProjectTemplateToV2(
+    templateId: string,
+    templateVersion: string,
+    projectName: string,
+    ref: LocationRef,
+    user?: UserIdentity
+  ): Promise<LocationRef> {
+    try {
+      // user storage maybe diff from template storage
+      const dstStorage = StorageService.getStorageClient(ref.storageId, user);
+      const dstDir = Path.resolve(ref.path);
+      if (await dstStorage.exists(dstDir)) {
+        log('Failed copying template to %s', dstDir);
+        throw new Error('already have this folder, please give another name');
       }
-      const { stderr: initErr2 } = await execAsync(`dotnet new ${template.id}`, {
-        cwd: destinationPath,
-      });
-      if (initErr2) {
-        throw new Error(initErr2);
+
+      mkdirSync(dstDir, { recursive: true });
+
+      // find selected template
+      const npmPackageName = templateId;
+      const generatorName = npmPackageName.toLowerCase().replace('generator-', '');
+
+      const remoteTemplateAvailable = await this.installRemoteTemplate(generatorName, npmPackageName, templateVersion);
+
+      if (remoteTemplateAvailable) {
+        await this.instantiateRemoteTemplate(generatorName, dstDir, projectName);
       }
```

Comment:
do we need an else here?

Reply:
maybe we should return true / false from installRemoteTemplate instead of using throw/catch?

lets pick one instead of both.

Reply:
Handled